src/lib.rs

```diff
@@ -8,12 +8,16 @@ pub type QueryParams<'a> = Vec<QueryParam<'a>>;
 /// ```
 /// extern crate querystring;
 ///
-/// assert_eq!(querystring::stringify(vec![("foo", "bar"), ("baz", "qux")]), "foo=bar&baz=qux&");
+/// assert_eq!(querystring::stringify(vec![("foo", "bar"), ("baz", "qux")]), "foo=bar&baz=qux");
 /// ```
 pub fn stringify(query: QueryParams) -> String {
-    query.iter().fold(String::new(), |acc, &tuple| {
-        acc + tuple.0 + "=" + tuple.1 + "&"
-    })
+    query
+        .iter()
+        .fold(String::new(), |acc, &tuple| {
+            acc + tuple.0 + "=" + tuple.1 + "&"
+        })
+        .trim_end_matches('&')
+        .to_string()
```

Comment:
I'm not 100% sure, but I think this might add an additional allocation.

It seemed like the best approach given the possibility space I could think of; not sure if there's room/a need to optimize further.